### DISCUSSION TITLE
I can't get my Pro 8 to work

### DISCUSSION THREAD

**Discussion by YosypC** [linux-surface/linux-surface#1157] [Q&A]
Hi y'all, I'm installing for the first time Arch on my Pro 8. I followed the wiki but I have many issues:

1. Bluetooth doesn't work (trying to switch it on on GNOME reverts it back to off)
2. Screen rotation doesn't work. I have no idea what tablet mode is since I've also never been able to show the screen keyboard.
3. The pen doesn't work. It's only recognized as a pointer and works so bad (for example, I try dragging, I stop dragging, and it doesn't stop pressing the "left mouse button" despite using the touchpad; it only stops when I press the touchpad).
4. Waking up from standby is tedious: pressing the power button seems to do nothing, clicking my mouse (which is powered after I press it) does nothing, and the type cover only works when replugged when it finally wakes up on its own.
5. The touchscreen worked fine although I have no way to change gestures. Now it decided to stop working entirely for no reason.
The wacom settings tab has disappeared.

lsmod output:
```
Module                  Size  Used by
snd_hda_codec_hdmi     94208  1
hid_multitouch         40960  0
snd_sof_pci_intel_tgl    16384  0
ccm                    20480  6
snd_sof_intel_hda_common   229376  1 snd_sof_pci_intel_tgl
algif_aead             16384  0
soundwire_intel        61440  1 snd_sof_intel_hda_common
cbc                    16384  0
soundwire_generic_allocation    16384  1 soundwire_intel
soundwire_cadence      45056  1 soundwire_intel
des_generic            16384  0
snd_sof_intel_hda      24576  1 snd_sof_intel_hda_common
libdes                 24576  1 des_generic
snd_sof_pci            24576  2 snd_sof_intel_hda_common,snd_sof_pci_intel_tgl
snd_sof_xtensa_dsp     20480  1 snd_sof_intel_hda_common
ecb                    16384  0
hid_sensor_gyro_3d     20480  0
hid_sensor_incl_3d     20480  0
snd_sof               389120  3 snd_sof_pci,snd_sof_intel_hda_common,snd_sof_intel_hda
hid_sensor_rotation    20480  0
hid_sensor_magn_3d     20480  0
hid_sensor_accel_3d    20480  0
hid_sensor_als         20480  0
hid_sensor_trigger     20480  13 hid_sensor_gyro_3d,hid_sensor_incl_3d,hid_sensor_als,hid_sensor_accel_3d,hid_sensor_magn_3d,hid_sensor_rotation
snd_sof_utils          20480  1 snd_sof
industrialio_triggered_buffer    16384  1 hid_sensor_trigger
snd_soc_hdac_hda       28672  1 snd_sof_intel_hda_common
kfifo_buf              16384  1 industrialio_triggered_buffer
snd_hda_ext_core       36864  3 snd_sof_intel_hda_common,snd_soc_hdac_hda,snd_sof_intel_hda
intel_rapl_msr         20480  0
algif_skcipher         16384  0
snd_soc_acpi_intel_match    77824  2 snd_sof_intel_hda_common,snd_sof_pci_intel_tgl
hid_sensor_iio_common    24576  7 hid_sensor_gyro_3d,hid_sensor_trigger,hid_sensor_incl_3d,hid_sensor_als,hid_sensor_accel_3d,hid_sensor_magn_3d,hid_sensor_rotation
uinput                 20480  0
intel_rapl_common      36864  1 intel_rapl_msr
snd_soc_acpi           16384  2 snd_soc_acpi_intel_match,snd_sof_intel_hda_common
soundwire_bus         135168  3 soundwire_intel,soundwire_generic_allocation,soundwire_cadence
industrialio          126976  9 industrialio_triggered_buffer,hid_sensor_gyro_3d,hid_sensor_trigger,hid_sensor_incl_3d,kfifo_buf,hid_sensor_als,hid_sensor_accel_3d,hid_sensor_magn_3d,hid_sensor_rotation
intel_tcc_cooling      16384  0
hid_sensor_custom      28672  0
x86_pkg_temp_thermal    20480  0
snd_soc_core          434176  4 soundwire_intel,snd_sof,snd_sof_intel_hda_common,snd_soc_hdac_hda
cmac                   16384  0
hid_sensor_hub         32768  9 hid_sensor_gyro_3d,hid_sensor_trigger,hid_sensor_iio_common,hid_sensor_incl_3d,hid_sensor_als,hid_sensor_accel_3d,hid_sensor_magn_3d,hid_sensor_rotation,hid_sensor_custom
joydev                 28672  0
mousedev               24576  0
md4                    16384  0
snd_hda_codec_realtek   196608  1
intel_powerclamp       16384  0
snd_compress           32768  1 snd_soc_core
surface_platform_profile    16384  0
iwlmvm                618496  0
algif_hash             16384  0
snd_hda_codec_generic   114688  1 snd_hda_codec_realtek
surface_hid            16384  0
platform_profile       16384  1 surface_platform_profile
surface_aggregator_hub    20480  0
surface_hid_core       16384  1 surface_hid
surface_aggregator_tabletsw    24576  0
surface_battery        24576  0
surface_charger        20480  0
ledtrig_audio          16384  1 snd_hda_codec_generic
ac97_bus               16384  1 snd_soc_core
af_alg                 36864  3 algif_hash,algif_skcipher,algif_aead
coretemp               20480  0
snd_pcm_dmaengine      20480  1 snd_soc_core
mac80211             1503232  1 iwlmvm
kvm_intel             458752  0
8250_dw                24576  0
snd_hda_intel          65536  3
snd_intel_dspcfg       36864  3 snd_hda_intel,snd_sof,snd_sof_intel_hda_common
libarc4                16384  1 mac80211
kvm                  1327104  1 kvm_intel
snd_intel_sdw_acpi     20480  2 snd_sof_intel_hda_common,snd_intel_dspcfg
irqbypass              16384  1 kvm
crct10dif_pclmul       16384  1
crc32_pclmul           16384  0
btusb                  77824  0
snd_hda_codec         217088  6 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec_realtek,snd_soc_hdac_hda,snd_sof_intel_hda
polyval_clmulni        16384  0
btrtl                  28672  1 btusb
polyval_generic        16384  1 polyval_clmulni
gf128mul               16384  1 polyval_generic
btbcm                  24576  1 btusb
ghash_clmulni_intel    16384  0
iwlwifi               548864  1 iwlmvm
sha512_ssse3           49152  0
snd_hda_core          139264  9 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hda_intel,snd_hda_ext_core,snd_hda_codec,snd_hda_codec_realtek,snd_sof_intel_hda_common,snd_soc_hdac_hda,snd_sof_intel_hda
aesni_intel           401408  4
btintel                57344  1 btusb
snd_hwdep              20480  1 snd_hda_codec
ov5693                 28672  0
ov13858                24576  0
crypto_simd            16384  1 aesni_intel
pmt_telemetry          16384  0
snd_pcm               200704  11 snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec,soundwire_intel,snd_sof,snd_sof_intel_hda_common,snd_compress,snd_soc_core,snd_sof_utils,snd_hda_core,snd_pcm_dmaengine
mei_pxp                20480  0
cryptd                 32768  2 crypto_simd,ghash_clmulni_intel
btmtk                  16384  1 btusb
mei_hdcp               28672  0
v4l2_fwnode            32768  2 ov5693,ov13858
intel_skl_int3472_tps68470    20480  0
pmt_class              16384  1 pmt_telemetry
gpio_keys              28672  0
intel_lpss_pci         28672  2
snd_timer              53248  1 snd_pcm
tps68470_regulator     16384  0
surface_gpe            24576  0
rapl                   20480  0
cfg80211             1286144  3 iwlmvm,iwlwifi,mac80211
bluetooth            1085440  6 btrtl,btmtk,btintel,btbcm,btusb
spi_nor               135168  0
v4l2_async             32768  3 v4l2_fwnode,ov5693,ov13858
mei_me                 61440  2
intel_cstate           24576  0
intel_uncore          258048  0
mtd                   110592  3 spi_nor
pcspkr                 16384  0
snd                   151552  17 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hwdep,snd_hda_intel,snd_hda_codec,snd_hda_codec_realtek,snd_sof,snd_timer,snd_compress,snd_soc_core,snd_pcm
intel_lpss             16384  1 intel_lpss_pci
clk_tps68470           16384  0
ecdh_generic           16384  1 bluetooth
videodev              372736  4 v4l2_async,v4l2_fwnode,ov5693,ov13858
vfat                   24576  1
thunderbolt           495616  0
soundcore              16384  1 snd
mei                   192512  5 mei_hdcp,mei_pxp,mei_me
rfkill                 40960  9 iwlmvm,bluetooth,cfg80211
surfacepro3_button     16384  0
intel_vsec             20480  0
ithc                   65536  0
idma64                 20480  0
igen6_edac             32768  0
fat                   102400  1 vfat
mc                     90112  4 v4l2_async,videodev,ov5693,ov13858
intel_skl_int3472_discrete    28672  0
surface_aggregator_registry    16384  0
surface_hotplug        20480  0
soc_button_array       24576  0
surface_aggregator    188416  8 surface_battery,surface_charger,surface_aggregator_hub,surface_aggregator_tabletsw,surface_platform_profile,surface_hid_core,surface_hid,surface_aggregator_registry
acpi_pad               24576  0
acpi_tad               20480  0
mac_hid                16384  0
pkcs8_key_parser       16384  0
crypto_user            20480  0
fuse                  204800  5
loop                   40960  0
dm_mod                225280  0
ip_tables              40960  0
x_tables               61440  1 ip_tables
ext4                 1163264  1
crc32c_generic         16384  0
crc16                  16384  2 bluetooth,ext4
mbcache                16384  1 ext4
jbd2                  217088  1 ext4
i915                 3850240  74
i2c_algo_bit           20480  1 i915
drm_buddy              20480  1 i915
intel_gtt              28672  1 i915
nvme                   65536  3
drm_display_helper    208896  1 i915
spi_intel_pci          16384  0
crc32c_intel           24576  2
nvme_core             237568  5 nvme
xhci_pci               24576  0
spi_intel              36864  1 spi_intel_pci
cec                    86016  2 drm_display_helper,i915
xhci_pci_renesas       24576  1 xhci_pci
video                  73728  1 i915
ttm                   102400  1 i915
nvme_common            24576  1 nvme_core
wmi                    45056  1 video
```

uname -a output:
```
Linux YOSYP-SURFACE 6.3.6-arch1-1-surface #1 SMP PREEMPT_DYNAMIC Tue, 06 Jun 2023 21:50:30 +0000 x86_64 GNU/Linux

```

**Comment by qzed** [comment]
> 1. Bluetooth doesn't work (trying to switch it on on GNOME reverts it back to off)

Did you enable `bluetooth.service`? What's the output of `systemctl status bluetooth.service`?

> 2. Screen rotation doesn't work. I have no idea what tablet mode is since I've also never been able to show the screen keyboard.

Do you have `iio-sensor-proxy` installed? And do you have a typecover attached when trying this? If you have one, can you try flipping it fully back and see if that helps?

> 3. The pen doesn't work. It's only recognized as a pointer and works so bad (for example, I try dragging, I stop dragging, and it doesn't stop pressing the "left mouse button" despite using the touchpad; it only stops when I press the touchpad).

Re pen being recognized as pointer: This depends on the application. The application explicitly needs to support pen/stylus input, otherwise it will fall back to being handled as a simple mouse pointer input device.

Re dragging issues: You can try increasing this value in the iptsd config: https://github.com/linux-surface/iptsd/blob/master/etc/iptsd.conf#L149 (`/etc/iptsd.conf`). This is  a trade-off between pen-position and pen-button, so try to increase it until the issues stop appearing but not as much that the primary pen button isn't recognized any more.

> 4. Waking up from standby is tedious: pressing the power button seems to do nothing, clicking my mouse (which is powered after I press it) does nothing, and the type cover only works when replugged when it finally wakes up on its own.

Can you post a dmesg log after suspend?

> 5. The touchscreen worked fine although I have no way to change gestures. Now it decided to stop working entirely for no reason.

When did it stop working? E.g. randomly, after suspend, or after an update?

**Comment by YosypC** [comment]
> > 1. Bluetooth doesn't work (trying to switch it on on GNOME reverts it back to off)
> 
> Did you enable `bluetooth.service`? What's the output of `systemctl status bluetooth.service`?
> 

Worked!

> > 2. Screen rotation doesn't work. I have no idea what tablet mode is since I've also never been able to show the screen keyboard.
> 
> Do you have `iio-sensor-proxy` installed? And do you have a typecover attached when trying this? If you have one, can you try flipping it fully back and see if that helps?
> 

Worked! What did I miss about the installation? I can't find "iio-sensor-proxy" mentioned anywhere in the guides. I also can't see any button to disable screen rotation.

> > 3. The pen doesn't work. It's only recognized as a pointer and works so bad (for example, I try dragging, I stop dragging, and it doesn't stop pressing the "left mouse button" despite using the touchpad; it only stops when I press the touchpad).
> 
> Re pen being recognized as pointer: This depends on the application. The application explicitly needs to support pen/stylus input, otherwise it will fall back to being handled as a simple mouse pointer input device.
> 
> Re dragging issues: You can try increasing this value in the iptsd config: https://github.com/linux-surface/iptsd/blob/master/etc/iptsd.conf#L149 (`/etc/iptsd.conf`). This is a trade-off between pen-position and pen-button, so try to increase it until the issues stop appearing but not as much that the primary pen button isn't recognized any more.
> 

[This video ](https://imgur.com/a/L78K9Hg)explains the behavior better. I also want to note that the Wacom tablet setting is empty, and by using a libwacom command (which I don't remember) my device isn't listed in it:in the libwacom-surface there's .patch files until the Pro 7.

> > 4. Waking up from standby is tedious: pressing the power button seems to do nothing, clicking my mouse (which is powered after I press it) does nothing, and the type cover only works when replugged when it finally wakes up on its own.
> 
> Can you post a dmesg log after suspend?
> 

[dmesg.zip](https://github.com/linux-surface/linux-surface/files/11705261/dmesg.zip)
This time it recovered fine pressing the power button, I'll experiment a bit more with the keyboard once I solve the issue that, after the first wakeup, I had to detach and reattach the Cover in order to be correctly recognized.

> > 5. The touchscreen worked fine although I have no way to change gestures. Now it decided to stop working entirely for no reason.
> 
> When did it stop working? E.g. randomly, after suspend, or after an update?

I updated everything time before. I tinkered a bit with the device, but I am sure it was the standby. After a reboot from yesterday, it's now working.

**Reply by qzed** [reply]
> > > 2. Screen rotation doesn't work. I have no idea what tablet mode is since I've also never been able to show the screen keyboard.
> > 
> > Do you have `iio-sensor-proxy` installed? And do you have a typecover attached when trying this? If you have one, can you try flipping it fully back and see if that helps?
> 
> Worked! What did I miss about the installation? I can't find "iio-sensor-proxy" mentioned anywhere in the guides. I also can't see any button to disable screen rotation.

It's not mentioned in the installation because it isn't really Surface specific. Any device with a screen that can be rotated based on some sensor needs that. For that reason, most distros (Fedora, Ubuntu, ...) usually install it out of the box (pretty similar to distros enabling bluetooth.service out of the box as well). So it's mostly an Arch thing...

> > > 3. The pen doesn't work. It's only recognized as a pointer and works so bad (for example, I try dragging, I stop dragging, and it doesn't stop pressing the "left mouse button" despite using the touchpad; it only stops when I press the touchpad).
> > 
> > Re pen being recognized as pointer: This depends on the application. The application explicitly needs to support pen/stylus input, otherwise it will fall back to being handled as a simple mouse pointer input device.
> > Re dragging issues: You can try increasing this value in the iptsd config: https://github.com/linux-surface/iptsd/blob/master/etc/iptsd.conf#L149 (`/etc/iptsd.conf`). This is a trade-off between pen-position and pen-button, so try to increase it until the issues stop appearing but not as much that the primary pen button isn't recognized any more.
> 
> [This video ](https://imgur.com/a/L78K9Hg)explains the behavior better. I also want to note that the Wacom tablet setting is empty, and by using a libwacom command (which I don't remember) my device isn't listed in it:in the libwacom-surface there's .patch files until the Pro 7.

This looks like a Gnome bug. Potentially https://gitlab.gnome.org/GNOME/mutter/-/issues/2773.

> > > 4. Waking up from standby is tedious: pressing the power button seems to do nothing, clicking my mouse (which is powered after I press it) does nothing, and the type cover only works when replugged when it finally wakes up on its own.
> > 
> > 
> > Can you post a dmesg log after suspend?
> 
> [dmesg.zip](https://github.com/linux-surface/linux-surface/files/11705261/dmesg.zip) This time it recovered fine pressing the power button, I'll experiment a bit more with the keyboard once I solve the issue that, after the first wakeup, I had to detach and reattach the Cover in order to be correctly recognized.

Thanks. Looks like something with device suspend ordering is going wrong. Could you please open up a new issue for that and also post the zip there?

> > > 5. The touchscreen worked fine although I have no way to change gestures. Now it decided to stop working entirely for no reason.
> > 
> > 
> > When did it stop working? E.g. randomly, after suspend, or after an update?
> 
> I updated everything time before. I tinkered a bit with the device, but I am sure it was the standby. After a reboot from yesterday, it's now working.

I think there's a similar issue in https://github.com/linux-surface/linux-surface/issues/1051.

**Comment by YosypC** [comment]
> I also want to note that the Wacom tablet setting is empty, and by using a libwacom command (which I don't remember) my device isn't listed in it:in the libwacom-surface there's .patch files until the Pro 7.

I ran "libwacom-list-local-devices" and I got:

/dev/input/event15 is a tablet but not supported by libwacom
/dev/input/event11 is a tablet but not supported by libwacom
Failed to find any devices known to libwacom.

Should I open an Issue?

**Reply by qzed** [reply]
We're currently missing a tablet data file for the SP8. Please open a new issue at https://github.com/linux-surface/libwacom-surface.

**Reply by YosypC** [reply]
[done!](https://github.com/linux-surface/libwacom-surface/issues/13)
